app/models/route_set.rb

```diff
@@ -67,7 +67,9 @@ def register!
         register_gone_route(route)
       end
     else
-      register_rendering_app
+      if SET_ROUTER_BACKEND_ADDRESS_ON_PUBLISH
+        register_rendering_app
+      end
```

Comment:
This looks good. It'd be good to have a test with it though

I imagine something like: 

```
    it "doesn't register a backend when NO_SET_ROUTER_BACKEND_ADDRESS_ON_PUBLISH environment variable is set" do
        ClimateControl.modify NO_SET_ROUTER_BACKEND_ADDRESS_ON_PUBLISH: "true" do
            backend_request = stub_request(:put, %r{\A#{GdsApi::TestHelpers::Router::ROUTER_API_ENDPOINT}/backends/})
            @item.register_routes
            expect(backend_request).not_to have_been_requested
        end
    end
```

Reply:
@kevindew I'm not quite seeing how the proposed test would invoke the initializer. (That's partly why I didn't commit the bodgy one-off test that I wrote, though also I considered that it wasn't worth the overhead of maintaining a test just for a feature flag)

Reply:
(just finished reading the rest of the comments - that test makes sense if we're reading the env var in the method)